problem_builder/slider.py

```diff
@@ -0,0 +1,157 @@
+        scope=Scope.content,
+        default="",
+        multiline_editor=True,
+    )
+
+    student_value = Float(
+        # The value selected by the student
+        default=None,
+        scope=Scope.user_state,
+    )
+
+    editable_fields = ('min_label', 'max_label', 'display_name', 'question', 'show_title')
+
+    def mentoring_view(self, context):
+        """ Main view of this block """
+        context = context.copy() if context else {}
+        context['question'] = self.question
+        context['slider_id'] = 'pb-slider-{}'.format(uuid.uuid4().hex[:20])
+        context['initial_value'] = int(self.student_value*100) if self.student_value is not None else 50
+        context['min_label'] = self.min_label
+        context['max_label'] = self.max_label
+        context['title'] = self.display_name_with_default
+        context['hide_header'] = context.get('hide_header', False) or not self.show_title
+        context['instructions_string'] = self._("Select a value from {min_label} to {max_label}").format(
+            min_label=self.min_label, max_label=self.max_label
+        )
+        html = loader.render_template('templates/html/slider.html', context)
+
+        fragment = Fragment(html)
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/slider.js'))
+        fragment.initialize_js('SliderBlock')
+        return fragment
+
+    def student_view(self, context=None):
+        """ Normal view of this XBlock, identical to mentoring_view """
+        return self.mentoring_view(context)
+
+    def author_view(self, context):
+        """
+        Add some HTML to the author view that allows authors to see the ID of the block, so they
+        can refer to it in other blocks such as Plot blocks.
+        """
+        context['hide_header'] = True  # Header is already shown in the Studio wrapper
+        fragment = self.student_view(context)
+        fragment.add_content(loader.render_template('templates/html/slider_edit_footer.html', {
+            "url_name": self.url_name
+        }))
+        return fragment
+
+    def get_last_result(self):
+        """ Return the current/last result in the required format """
+        if self.student_value is None:
+            return {}
+        return {
+            'submission': self.student_value,
+            'status': 'correct',
+            'tips': [],
+            'weight': self.weight,
+            'score': 1,
+        }
+
+    def get_results(self, _previous_result_unused=None):
+        """ Alias for get_last_result() """
+        return self.get_last_result()
+
+    def submit(self, value):
+        log.debug(u'Received Slider submission: "%s"', value)
+        if value < 0 or value > 1:
+            return {}  # Invalid
+        self.student_value = value
+        if sub_api:
+            # Also send to the submissions API:
+            sub_api.create_submission(self.student_item_key, {'value': value, })
```

Comment:
@bradenmacdonald Nit: Since the dict that's passed to `create_submission` fits on one line I'd probably remove the trailing comma and space: `{'value': value, }` ⟶ `{'value': value}`